src/coreclr/vm/interoplibinterface_comwrappers.cpp

```diff
@@ -1169,11 +1169,7 @@ namespace InteropLibImports
         bool isValid = false;
         ::OBJECTHANDLE objectHandle = static_cast<::OBJECTHANDLE>(handle);
 
-        {
-            // Switch to cooperative mode so the handle can be safely inspected.
-            GCX_COOP_THREAD_EXISTS(GET_THREAD());
-            isValid = ObjectFromHandle(objectHandle) != NULL;
-        }
+        isValid = ObjectHandleIsNull(objectHandle) == TRUE;
```

Comment:
```suggestion
        isValid = ObjectHandleIsNull(objectHandle) != FALSE;
```
It is bug prone to compare `BOOL` with `TRUE` and some static analysis tools will flag it as a problem.
For example https://pvs-studio.com/en/docs/warnings/v676/